native/spark-expr/src/math_funcs/internal/make_decimal.rs

```diff
@@ -40,18 +40,21 @@ pub fn spark_make_decimal(
             ))),
             sv => internal_err!("Expected Int64 but found {sv:?}"),
         },
-        ColumnarValue::Array(a) => {
-            let arr = a.as_primitive::<Int64Type>();
-            let mut result = Decimal128Builder::new();
-            for v in arr.into_iter() {
-                result.append_option(long_to_decimal(&v, precision))
-            }
-            let result_type = DataType::Decimal128(precision, scale);
+        ColumnarValue::Array(a) => match a.data_type() {
+            DataType::Int64 => {
```

Comment:
should we also support Int32? 🤔

Reply:
for this PR, I just wanted to fall back to avoid crashing. We could choose to add support in the future though